.pre-commit-config.yaml

```diff
@@ -0,0 +1,43 @@
+# Exclude files which shouldn't be modified by pre-commit hooks.
+exclude: |
+  (?x)^(
+    .*\.sln| # Expected to contain a BOM
+    .*\.csproj| # Expected to contain a BOM
+    .*\.patch| # Expected to contrain trailing whitespace
+    .*\.pot|
+    .*\.po|
+    thirdparty/.*| # Third-party files
+    platform/android/java/lib/src/com/google.*| # Third-party files
+    .*-so_wrap\..* # Generated code
+  )
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
+    hooks:
+      - id: fix-byte-order-marker
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+
+      - id: mixed-line-ending
+        args: [--fix=lf]
+
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.1.1
+    hooks:
+      - id: clang-format
+
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args: [-l 120]
+
+  - repo: local
+    hooks:
+    - id: makerst
+      name: Check XML class reference syntax and validity
+      language: system
+      pass_filenames: false
+      entry: python doc/tools/makerst.py doc/classes modules --dry-run
```

Comment:
```suggestion
      entry: python3 doc/tools/makerst.py doc/classes modules --dry-run
```

Reply:
Might need testing on Windows as typically the installed binary there is `python.exe` for Python 3.
And apparently @BastiaanOlij has a setup where it's `py.exe` :scream_cat:

Reply:
Indeed, Windows seems to use `python.exe`, at least when it's installed from www.python.org. And probably won't run the script directly if `python` is removed from the command (which is also a valid option on macOS). I'll check whether it works on Windows at all a bit later.

Reply:
> I'll check whether it works on Windows at all a bit later.

Seems to work fine on Windows.

> probably won't run the script directly if python is removed

Windows shell is capable of running python scripts directly, but `pre-commit` is not.

Reply:
> 
> 
> Might need testing on Windows as typically the installed binary there is `python.exe` for Python 3.
> And apparently @BastiaanOlij has a setup where it's `py.exe` 🙀

Yeah but the thing is this stupid STUB thing that Windows does so if you try and run python, instead of it running your installed python, it will run this little program that redirects you to MSVC and tells it to install its version of Python. Like WTF.. 

py.exe worked because that didn't have a stub.

Can't remember the suggestion but someone told me how to turn that off and now the python executable in my python install works

Reply:
pre-commit has a config option [default_language_version](https://pre-commit.com/#top_level-default_language_version). Setting it to `python3` is enough to use Python 3 even with `python` executable name

minimal reproducible example:
```
default_language_version:
  python: python3

repos:
-   repo: local
    hooks:
    -   id: check-version
        name: check python version
        language: python
        entry: python --version
        files: ".*"
```

```
$ pre-commit run --verbose
>>> Python 3.10.6
```

output with `python: python2`:
```
>>> Python 2.7.17
```